x-pack/plugins/canvas/canvas_plugin_src/renderers/dropdown_filter/index.tsx

```diff
@@ -20,20 +21,40 @@ interface Config {
    * @default []
    */
   choices: string[];
+  filterGroup: string;
 }
 
+const MATCH_ALL = '%%CANVAS_MATCH_ALL%%';
+
+const getFilterValue = (filterExpression: string) => {
+  if (filterExpression === '') {
+    return MATCH_ALL;
+  }
+
+  const filterAST = fromExpression(filterExpression);
+  return get(filterAST, 'chain[0].arguments.value[0]', MATCH_ALL) as string;
+};
+
 export const dropdownFilter: RendererFactory<Config> = () => ({
   name: 'dropdown_filter',
   displayName: 'Dropdown filter',
   help: 'A dropdown from which you can select values for an "exactly" filter',
   reuseDomNode: true,
   height: 50,
   render(domNode, config, handlers) {
-    const { column, choices } = config;
-    let value = '%%CANVAS_MATCH_ALL%%';
-    if (handlers.getFilter() !== '') {
-      const filterAST = fromExpression(handlers.getFilter());
-      value = get(filterAST, 'chain[0].arguments.value[0]');
+    const filterExpression = handlers.getFilter();
+
+    if (filterExpression !== '') {
+      // NOTE: setFilter() will cause a data refresh, avoid calling unless required
+      // compare expression and filter, update filter if needed
+      const { changed, newAst } = syncFilterExpression(config, filterExpression, [
```

Comment:
This is unfortunately becoming a common pattern, where expression args need to be synced with the filter expression at render time. I made a helper but I'm not sure how to abstract it better ☹️